src/components/application_manager/src/display_capabilities_builder.cc

```diff
@@ -152,7 +154,27 @@ bool DisplayCapabilitiesBuilder::IsWaitingForWindowCapabilities(
 void DisplayCapabilitiesBuilder::ResetDisplayCapabilities() {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock lock(display_capabilities_lock_);
-  display_capabilities_.reset();
+  for (auto& window_id : window_ids_to_resume_) {
+    if (kDefaultWindowID != window_id) {
+      window_ids_to_resume_.erase(window_id);
+    }
+  }
+
+  if (display_capabilities_) {
+    auto cur_window_caps =
+        (*display_capabilities_)[0][strings::window_capabilities].asArray();
+    if (cur_window_caps) {
+      std::remove_if(cur_window_caps->begin(),
+                     cur_window_caps->end(),
+                     [](const smart_objects::SmartObject& item) {
+                       const WindowID window_id =
+                           item.keyExists(strings::window_id)
+                               ? item[strings::window_id].asInt()
+                               : kDefaultWindowID;
+                       return kDefaultWindowID != window_id;
```

Comment:
This function return bool? 
Why do we need this intemediate variable window_id?
Why not just `return item.keyExists(strings::window_id)`?

Reply:
@LuxoftAKutsan window_id is an optional parameter in cabapilities, so we have actually 3 cases to cover:
1. Return false when `window_id` key does not exist
2. Return false when `window_id` key exists and equal to `kDefaultWindowID`
3. Return true when `window_id` key exists and not equal to `kDefaultWindowID`

Reply:
@LuxoftAKutsan actually, it can be simplified according to third condition. I've changed this lambda and removed extra variable in e963dbd